app/javascript/onboarding/__tests__/Onboarding.test.jsx

```diff
@@ -1,416 +1,186 @@
 import { h } from 'preact';
-import { deep } from 'preact-render-spy';
-import fetch from 'jest-fetch-mock';
+import { render, waitForElement } from '@testing-library/preact';
 import { axe } from 'jest-axe';
+import fetch from 'jest-fetch-mock';
+import '@testing-library/jest-dom';
 
 import Onboarding from '../Onboarding';
-import ProfileForm from '../components/ProfileForm';
-import FollowTags from '../components/FollowTags';
-import FollowUsers from '../components/FollowUsers';
-
 global.fetch = fetch;
 
-// Corresponds to the order of slides declared in the Onboarding component.
```

Comment:
This file has the most changes and is split up, hence I thought I'd do a quick write up on the rationale and approach.

We were testing a lot of state in the Onboarding tests, however, Kent Dodds describes that the idea of testing with react-testing-library is to interact with the component as close to how the end user would, by firing events on rendered element. The user would not have insight into the state, but instead they would witness the effects of the state changes on the interface. 
Therefore, we've changed the approach of these tests as they proved difficult to easily convert.

The new approach is as follow:
We've split up Onboarding into multiple files where we test each individual step of the component (i.e 'IntroSlide', 'FollowTags', 'ProfileForm', 'FollowUsers', 'EmailPreferencesForm'). 

Each component has its own individual test where we exhaustively test how the isolated component behaves with its relevant props. (we can find these new test files here _app/javascript/onboarding/components/_tests/...._).

Thereafter, the onboarding file tests how the different components interact with each other. i.e. here we test how the components move forward and backward when we click on next or back or skip a step. At some point it will be beneficial to replace/compliment this file with an end-to-end test as well.